src/Pyz/Yves/ShopUi/Theme/default/components/organisms/side-drawer/side-drawer.ts

```diff
@@ -1,45 +1,71 @@
-import Component from 'ShopUi/models/component';
+import SideDrawerCore from 'ShopUi/components/organisms/side-drawer/side-drawer';
 
-export default class SideDrawer extends Component {
-    protected triggers: HTMLElement[];
-    protected containers: HTMLElement[];
-
-    protected readyCallback(): void {}
+export default class SideDrawer extends SideDrawerCore {
+    protected overlay: HTMLElement;
 
     protected init(): void {
-        this.triggers = <HTMLElement[]>Array.from(document.getElementsByClassName(this.triggerClassName));
-        this.containers = <HTMLElement[]>Array.from(document.getElementsByClassName(this.containerClassName));
-        this.mapEvents();
+        this.overlay = <HTMLElement>document.getElementsByClassName(this.overlayClassName)[0];
+
+        super.init();
     }
 
     protected mapEvents(): void {
-        this.triggers.forEach((trigger: HTMLElement) => {
-            trigger.addEventListener('click', (event: Event) => this.onTriggerClick(event));
-        });
-    }
+        super.mapEvents();
 
-    protected onTriggerClick(event: Event): void {
-        event.preventDefault();
-        this.toggle();
+        this.mapWindowResizeEvent();
     }
 
-    toggle(): void {
-        const isShown = !this.classList.contains(`${this.name}--show`);
-        this.classList.toggle(`${this.name}--show`, isShown);
-        this.containers.forEach((conatiner: HTMLElement) => {
-            conatiner.classList.toggle(this.lockedBodyClassName, isShown);
+    protected mapWindowResizeEvent(): void {
+        window.addEventListener('resize', () => {
+            if (!this.classList.contains(`${this.name}--show`)) {
+                return;
+            }
+
+            if (window.innerWidth >= this.overlayBreakpoint) {
+                this.overlay.classList.remove(`${this.overlayClassName}--show`);
+
+                return;
+            }
+
+            this.overlay.classList.add(`${this.overlayClassName}--show`);
```

Comment:
Can we solve it via css media queries?

Reply:
There are 2 breakpoints: 
- 768 for `account-navigation` which extends `side-drawer`
- 1024 for default `side-drawer`